packages/core/test/actions.test.ts

```diff
@@ -1682,6 +1682,32 @@ describe('purely defined actions', () => {
   });
 });
 
+describe('action groups', () => {
+  it('should execute actions provided as type strings in a group', () => {
```

Comment:
Awesome! Can you also add tests to support:

- Mixture of strings, objects, functions
- An action group that references another action that references that action group (edge-case)
- Some more edge-cases I can't think of right now

Reply:
Added some tests for groups of each type as well as a mixture

> An action group that references another action that references that action group (edge-case)

Would we just test that this throws a maximum call stack exceeded error for the time being? 

Was playing around with a way to track and prevent cyclical action calls last night because `pure` and `choose` can suffer from the same problem, but it feels pretty thorny

Reply:
> Would we just test that this throws a maximum call stack exceeded error for the time being?

While creating the action, you can have some variable in closure (like a `Set`) that keeps track of actions already added to the group, and if it encounters it again (`set.has(someActionType)`), you can detect that cycle.

Reply:
Just want to make sure I'm understanding. Are you thinking this `Set` would be within a cycle test or in the actual codebase?